mythril/analysis/modules/dependence_on_predictable_vars.py

```diff
@@ -14,174 +14,180 @@ def __init__(self):
-                                title="Dependence on predictable variable",
-                                _type="Warning",
-                                description=description,
-                                swc_id=PREDICTABLE_VARS_DEPENDENCE,
-                                gas_used=(
-                                    call.state.mstate.min_gas_used,
-                                    call.state.mstate.max_gas_used,
-                                ),
-                            )
-                            issues.append(issue)
-                            break
-                    else:
-                        r = re.search(r"storage_([a-z0-9_&^]+)", str(constraint))
-                        if r:  # block.blockhash(storage_0)
-
-                            """
-                            We actually can do better here by adding a constraint blockhash_block_storage_0 == 0
-                            and checking model satisfiability. When this is done, severity can be raised
-                            from 'Informational' to 'Warning'.
-                            Checking that storage at given index can be tainted is not necessary, since it usually contains
-                            block.number of the 'commit' transaction in commit-reveal workflow.
-                            """
-
-                            index = r.group(1)
-                            if index and self.solve(call):
-                                description += (
-                                    "block.blockhash() is calculated using a value from storage "
-                                    "at index {}".format(index)
-                                )
-                                issue = Issue(
-                                    contract=call.node.contract_name,
-                                    function_name=call.node.function_name,
-                                    address=address,
-                                    bytecode=call.state.environment.code.bytecode,
-                                    title="Dependence on predictable variable",
-                                    _type="Informational",
-                                    description=description,
-                                    swc_id=PREDICTABLE_VARS_DEPENDENCE,
-                                    gas_used=(
-                                        call.state.mstate.min_gas_used,
-                                        call.state.mstate.max_gas_used,
-                                    ),
-                                )
-                                issues.append(issue)
-                                break
+                    break
+            else:
+                r = re.search(r"storage_([a-z0-9_&^]+)", str(constraint))
+                if r:  # block.blockhash(storage_0)
+
+                    """
+                    We actually can do better here by adding a constraint blockhash_block_storage_0 == 0
```

Comment:
Why don't we take the opportunity refactoring the module to do this?

Reply:
We could, but unfortunately the storage is concrete currently(previously it was unconstrained) so that should be changed before doing something here.We can change but we can't test it without changing that, so currently this entire if condition is redundant :cry: